version: '3'
services:
  nest_nginx:
    container_name: nest_nginx
    image: nginx:alpine
    ports:
      - "80:8000"
    networks:
      - nest
    volumes:
      - './docker/nginx/default.conf:/etc/nginx/conf.d/default.conf'
  nest_auth:
    container_name: nest_auth
    build:
      context: .
      dockerfile: ./apps/auth/Dockerfile
      target: development
    command: "pnpm run start:dev auth"
    ports:
      - "3001:3001"
    networks:
      - nest
    volumes:
      - .:/usr/src/app
    env_file:
      - ./apps/auth/.env
  nest_redis:
    container_name: nest_redis
    image: 'redis:alpine'
    ports:
      - '${FORWARD_REDIS_PORT:-6379}:6379'
    networks:
      - nest
    volumes:
      - './docker/data/redis:/data'
    healthcheck:
      test: [ "CMD", "redis-cli", "ping" ]
      retries: 3
      timeout: 5s
  nest_mongodb:
    container_name: nest_mongodb
    image: mongo:latest
    networks:
      - nest
    ports:
      - '27017:27017'
    volumes:
      - ./docker/data/mongodb:/data/db
    environment:
      MONGO_INITDB_ROOT_USERNAME: default
      MONGO_INITDB_ROOT_PASSWORD: secret

networks:
  nest:
    driver: bridge